### PR TITLE
[38-productlist-model] Adding ProductList data model and model tests

### PR DESCRIPTION
**Pull request by evpoe**
- Added the Product model and Association model
- Updated test_models with new cases to cover the Product table

Note: We already have 90% coverage but I have some Association model test cases that still need to be finished. These have been commented out at the bottom.

### CODE REVIEW

**Comment by CatherineRH** [comment]
See the 84% coverage. Do you want help with the test cases?

File: tests/test_models.py [hunk 1]
```diff
@@ -44,19 +44,39 @@ def _create_supplier(self):
...
+        supplier = Supplier(
+            name="Jim Jones",
+            address="123 Main Street, Anytown USA", 
+            email="jjones@gmail.com", 
+            phone_number="800-555-1212",
+            products=[]
+        )
+        association1 = Association(supplier_id = supplier.id, wholesale_price=999)
+        association.product = Product(name="Macbook")
+        association2 = Association(supplier_id = supplier.id, wholesale_price=999)
```
Comment (beafus) [inline tests/test_models.py:60]: why are these associations with the same values?

File: tests/test_models.py [hunk 2]
```diff
@@ -44,19 +44,39 @@ def _create_supplier(self):
...
+    def _create_supplier_with_product(self): 
+        supplier = Supplier(
+            name="Jim Jones",
+            address="123 Main Street, Anytown USA", 
+            email="jjones@gmail.com", 
+            phone_number="800-555-1212",
+            products=[]
+        )
+        association1 = Association(supplier_id = supplier.id, wholesale_price=999)
+        association.product = Product(name="Macbook")
```
Comment (beafus) [inline tests/test_models.py:59]: maybe you could use here you helper function? _create_product

**Review by beafus — COMMENTED** [review]
left some comments

File: tests/test_models.py [hunk 3]
```diff
@@ -236,9 +247,124 @@ def test_find_or_404_found(self):
...
+        self.assertNotEqual(product, None)
+        self.assertEqual(product.id, None)
+        self.assertEqual(product.name, "Macbook")
+
+
+    ######################################################################
+    #  A S S O C I A T I O N   T E S T   C A S E S
+    ######################################################################
+        
+    # def test_create_a_supplier_with_product(self):
```
Comment (beafus) [inline tests/test_models.py:348]: Lets uncomment this and add the missing tests for the association

**Review by beafus — CHANGES_REQUESTED** [review]

File: tests/test_models.py (hunk 1 above)
Comment (evpoe) [reply tests/test_models.py]: Cleaned this up

File: tests/test_models.py (hunk 3 above)
Comment (evpoe) [reply tests/test_models.py]: Added additional tests

File: tests/test_models.py (hunk 2 above)
Comment (evpoe) [reply tests/test_models.py]: Updated helper function

**Review by beafus — APPROVED** [review]

**Pull request merged**